fms2_io/include/gather_data_bc.inc

```diff
@@ -0,0 +1,256 @@
+          local_buf_r4_kind = 0.
+          isc = 1+i_glob; i_add=0; iec=1+i_glob; jsc=j_glob; j_add=0; jec=j_glob
+          i1=1; i2=1; j1=1; j2=1
+       else
+          !! In this case there is data in fileobj's root, so there is no need for the dummy data
+          if(fileobj%is_root) allocate(global_buf_r4_kind(i_glob, j_glob))
+          allocate(local_buf_r4_kind(size(vdata,1), size(vdata,2)))
+          local_buf_r4_kind = vdata
+       endif
+
+       call mpp_gather(isc+i_add, iec+i_add, jsc+j_add, jec+j_add, bc_info%pelist, &
+                          local_buf_r4_kind(i1:i2,j1:j2), &
+                          global_buf_r4_kind, fileobj%is_root)
+
+        deallocate(local_buf_r4_kind)
+       !> If you are on fileobj's root, calculate the checksum and save the gathered data in a buffer
+       if (fileobj%is_root) then
+           chksum_val = mpp_chksum(global_buf_r4_kind(1:i_glob,1:j_glob), (/mpp_pe()/))
+           allocate(bc_info%globaldata2d_r4(i_glob, j_glob))
+           bc_info%globaldata2d_r4=global_buf_r4_kind(1:i_glob,1:j_glob)
+           deallocate(global_buf_r4_kind)
+       endif
+    type is (real(kind=r8_kind))
+       !> If the fileobj's root does not have data for this variable
+       if (fileobj%is_root .and. .not. bc_info%data_on_file_root) then
+          !> Allocate global_buf_r8_kind to be one size bigger, global_buf_r8_kind(i_glob+1,,:) is just dummy data
+          allocate(global_buf_r8_kind(i_glob+1, j_glob))
+          !> Allocate a temp local buffer to the fileobj's root. This is needed because the data needs to be send
+          !! to the fileobj's root, but because fileobj's root doesn't have any data, we just create dummy data and
+          !! not write it later
+          allocate(local_buf_r8_kind(1,1))
+          local_buf_r8_kind = 0.
+          isc = 1+i_glob; i_add=0; iec=1+i_glob; jsc=j_glob; j_add=0; jec=j_glob
+          i1=1; i2=1; j1=1; j2=1
+       else
+          !! In this case there is data in fileobj's root, so there is no need for the dummy data
+          if(fileobj%is_root) allocate(global_buf_r8_kind(i_glob, j_glob))
+          allocate(local_buf_r8_kind(size(vdata,1), size(vdata,2)))
+          local_buf_r8_kind = vdata
+       endif
+
+       call mpp_gather(isc+i_add, iec+i_add, jsc+j_add, jec+j_add, bc_info%pelist, &
+                          local_buf_r8_kind(i1:i2,j1:j2), &
+                          global_buf_r8_kind, fileobj%is_root)
+       deallocate(local_buf_r8_kind)
+       !> If you are on fileobj's root, calculate the checksum and save the gathered data in a buffer
+       if (fileobj%is_root) then
+           chksum_val = mpp_chksum(global_buf_r8_kind(1:i_glob,1:j_glob), (/mpp_pe()/))
+           allocate(bc_info%globaldata2d_r8(i_glob, j_glob))
+           bc_info%globaldata2d_r8=global_buf_r8_kind(1:i_glob,1:j_glob)
+           deallocate(global_buf_r8_kind)
+       endif
+     class default
+        call error("unsupported type.")
```

Comment:
@uramirez8707 This error message needs to be updated

Reply:
Fixed in f0d7161f31478f7ef7f92b4eee505d409ef3b0eb